src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/FilterMutectCalls.java

```diff
@@ -0,0 +1,74 @@
+package org.broadinstitute.hellbender.tools.walkers.mutect;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.variantcontext.VariantContextBuilder;
+import htsjdk.variant.variantcontext.writer.VariantContextWriter;
+import htsjdk.variant.vcf.VCFFilterHeaderLine;
+import htsjdk.variant.vcf.VCFHeader;
+import htsjdk.variant.vcf.VCFHeaderLine;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.ArgumentCollection;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.VariantProgramGroup;
+import org.broadinstitute.hellbender.engine.*;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFHeaderLines;
+import org.broadinstitute.hellbender.utils.variant.GATKVariantContextUtils;
+
+import java.io.File;
+import java.util.*;
+
+@CommandLineProgramProperties(
+        summary = "Filter somatic SNPs and indels called by Mutect2",
+        oneLineSummary = "Filter somatic SNPs and indels called by Mutect2",
+        programGroup = VariantProgramGroup.class
+)
+public final class FilterMutectCalls extends VariantWalker {
+
+    @Argument(fullName= StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName=StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            doc="The output filtered VCF file", optional=false)
+    private final String outputVcf = null;
+
+    @ArgumentCollection
+    protected M2FiltersArgumentCollection MTFAC = new M2FiltersArgumentCollection();
+
+    private VariantContextWriter vcfWriter;
+
+    private final List<VariantContext> unfilteredCalls = new ArrayList<>();
+
+
+    @Override
+    public void onTraversalStart() {
+        final VCFHeader inputHeader = getHeaderForVariants();
+        final Set<VCFHeaderLine> headerLines = new HashSet<>(inputHeader.getMetaDataInSortedOrder());
+        Mutect2FilteringEngine.M_2_FILTER_NAMES.stream().map(GATKVCFHeaderLines::getFilterLine).forEach(headerLines::add);
+        headerLines.add(new VCFFilterHeaderLine(Mutect2FilteringEngine.ARTIFACT_IN_NORMAL_FILTER_NAME, "artifact_in_normal"));
+
+        final VCFHeader vcfHeader = new VCFHeader(headerLines, inputHeader.getGenotypeSamples());
+        vcfWriter = GATKVariantContextUtils.createVCFWriter(new File(outputVcf), getBestAvailableSequenceDictionary(), lenientVCFProcessing);
+        vcfWriter.writeHeader(vcfHeader);
+    }
+
+    @Override
+    public Object onTraversalSuccess() {
```

Comment:
Is there a reason it does all the work in onSuccess instead of streaming it during apply?